### ISSUE TITLE
Create a VS Code extension

### ISSUE THREAD

**Issue by filipw** [filipw/dotnet-script#197]
We could create a VS Code extension that would streamline some of the scripting stuff:

 - run script
 - run code block
 - init a script + `launch.json`
 - help with `omnisharp.json`?

We could even have a terminal-like feature in there to be used with REPL.
In the past I experimented a bit with this for scriptcs - here is the ancient repo https://github.com/filipw/vscode-scriptcs-runner

**Comment by seesharper** [comment]
Just to summarize what we've got already 

```
dotnet script init
```
Creates `Helloworld.csx`, `.vscode/launch.json` and `omnisharp.json`

It would be awesome to get a question asking to create these assets in VS Code in case they are missing. 

When it comes to running the script from within VS Code that is already there given that we have a `.vscode/launch.json` file created from `dotnet script init`

Debug Menu
* Start Debugging
* Start Without Debugging  
 
If we should want to do initialization from VS Code, we could simply call into what we already ship today.

**Comment by seesharper** [comment]
It would also be nice if we had a story for testing scripts. No idea how that would look like yet, though :)

**Comment by filipw** [comment]
Yeah the way I see it, most of the extension feature would simply call into dotnet-script (this is what many VS Code extensions do - simply call into the CLI tools). 
But being able to bootstrap everything from the editor level (command palette, shortcuts etc), as soon as you just create a CSX file would be great.

**Comment by seesharper** [comment]
Would it be possible to  hook into the editor to provide statement completion for NuGet packages?
```
#r "nuget: dropdown here to select packages"
```

**Comment by filipw** [comment]
yeah I guess so - while C# for VS Code doesn't do it for the time being, this extension does it for csproj https://github.com/tintoy/msbuild-project-tools-vscode

You could also imagine other features i.e. convert to full C# command line project etc.

**Comment by SlowLogicBoy** [comment]
https://marketplace.visualstudio.com/items?itemName=jmrog.vscode-nuget-package-manager  I use this for projects. However it only works with ctrl+shift+p, no drop down and it doesn't work with .csx There is also Paket through ionide

**Comment by migajek** [comment]
for me the most handy feature would be the possibility to run the content of the VS Code editor without creating any "project" (via `dotnet script init`)

just run VS code, create new file, type some code and run it. 

kind of like LINQPad ...

**Comment by seesharper** [comment]
@filipw Just briefly started to look into creating an extension for `dotnet-script`

I was thinking that the very first feature could be running `dotnet script init` to setup `launch.json` and `omnisharp.json`. 

We could do something similar to what happens when opening a standard .Net core console app in VS code

![image](https://user-images.githubusercontent.com/1034073/34152634-8d80dff8-e4af-11e7-9dd7-d06b0fd42871.png)

We could display something like 

> Required assets to debug and reference inline NuGet packages are missing from 'ScriptFolder'. Add them?

**Comment by ivanakcheurov** [comment]
_Guys, really great tool!_

Scripting C# is convenient in VSCode because of Code Completion but lacks good REPL.

Some of the immediate wishes for basic **REPL in VSCode**:
- to run code per line or selected block from the main code window in VSCode (ideally Ctrl-Enter like in RStudio, or at least F8 in Powershell ISE). 
- to execute ad-hoc code from VSCode terminal in the current session (like RStudio, Powershell ISE).
- to store all the session variables and functions in a session common to both the main code window and the terminal.

**Console** is convenient for quick "lemme check" REPL but needs Code Completion and Syntax Highlighting (like Powershell or bpython).

_Thank you for making dotnet-script such a consistent and modern tool, enjoying it!_

**Comment by casz** [comment]
I would very much need code completion to make it worth using dotnet script.

**Comment by duncanawoods** [comment]
You can use the [markdown-preview-enhanced](https://marketplace.visualstudio.com/items?itemName=shd101wyy.markdown-preview-enhanced) extension to execute code chunks with dotnet-script.

    ``` csharp {cmd="dotnet" args=["script"]}
    Console.WriteLine("Hello babe!");
    ```

**Comment by iSeiryu** [comment]
@ivanakcheurov This tool was able to achieve code completion and syntax highlight in the terminal https://github.com/waf/CSharpRepl. I really hope dotnet-script can do it too.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/dotnet-script/dotnet-script/issues/197#issue-279082302

@filipw, https://marketplace.visualstudio.com/items?itemName=Wivuu.dotnetscriptr&ssr=false#overview may be of use.
